### DISCUSSION TITLE
InstallerType: Wix vs Burn?

### DISCUSSION THREAD

**Discussion by cgutman** [microsoft/winget-cli#1965] [Q&A]
I see that there are both [Wix and Burn installer types](https://github.com/microsoft/winget-cli/blob/e997f3e394e66f2389f4f33f36c58b1c2768e975/src/AppInstallerRepositoryCore/Manifest/ManifestInstaller.cpp#L117-L118) present in `winget`, yet I'm unsure what the difference between the two would be. They seem to be handled identically everywhere in the code.

As far as I'm aware, the WiX tools can either output an exe, which would be a Burn-based bundled installer (aka `InstallerType: burn`), or a regular MSI (aka `InstallerType: msi`). I'm not sure what value `InstallerType: wix` has.

I see you have a [test for burn installers](https://github.com/microsoft/winget-cli/blob/e997f3e394e66f2389f4f33f36c58b1c2768e975/src/AppInstallerCLIE2ETests/TestData/Manifests/TestBurnInstaller.yaml) but not for Wix. On the flip side, your [manifest spec](https://github.com/microsoft/winget-cli/blob/master/doc/ManifestSpecv0.1.md) references `wix` but not `burn`.

Is one meant to be deprecated in favor of the other? Are they aliases?

**Comment by denelon** [comment]
I had to do some digging, and here is what I found:
https://heiswayi.nrird.com/installer-bootstrapping-with-burn

We use the knowledge of the installer type to pass the appropriate arguments for silent install. They appear to be used as an alias to one another. The WiX toolset includes burn, so I believe "wix" is the best value to use, and I'm not sure why the test is using "burn". @yao-msft might know more.
